src/standalone/Soil/soil_hydrology_parameterizations.jl

```diff
@@ -7,7 +7,108 @@ export volumetric_liquid_fraction,
+"""
+function albedo_from_moisture(
+    surface_liquid_fraction::FT,
+    albedo_dry::FT,
+    albedo_wet::FT,
+) where {FT}
+    return albedo_dry * (1 - surface_liquid_fraction) +
+           albedo_wet * surface_liquid_fraction
+end
+
+
+"""
+    update_albedo!(bc::AtmosDrivenFluxBC, p, soil_domain, model_parameters)
+
+Calculates and updates PAR and NIR albedo as a function of volumetric soil water content at
+the top of the soil. If the soil layers are very large, the water content of the top layer
+is used in the calclulation
+"""
+function update_albedo!(bc::AtmosDrivenFluxBC, p, soil_domain, model_parameters)
+    (;
+        ν,
+        θ_r,
+        PAR_albedo_dry,
+        NIR_albedo_dry,
+        PAR_albedo_wet,
+        NIR_albedo_wet,
+        top_depth,
+    ) = model_parameters
+    # ∫H_S_e_dz is the integral of effective saturation from (surface-top_depth) to surface
+    ∫H_S_e_dz = p.soil.sfc_θ_l
+    FT = eltype(soil_domain.fields.Δz_top)
+    # checks if there is at least 1 layer centered within the top soil depth
+    if maximum(soil_domain.fields.Δz_top) < top_depth
+        # ∫H_dz is integral of 1 from (surface-top_depth) to surface
+        ∫H_dz = p.soil.sfc_scratch
+        # calculate depths of layer centers
+        @. p.soil.sub_sfc_scratch =
+            soil_domain.fields.z_sfc - soil_domain.fields.z
+        # zero all centers lower than boundry, set everything above to one
+        @. p.soil.sub_sfc_scratch =
+            ClimaLand.heaviside.(
+                top_depth + sqrt(eps(FT)),
+                p.soil.sub_sfc_scratch,
+            )
+        ClimaCore.Operators.column_integral_definite!(
+            ∫H_dz,
+            p.soil.sub_sfc_scratch,
+        )
+        # calculate depths again
+        @. p.soil.sub_sfc_scratch =
+            soil_domain.fields.z_sfc - soil_domain.fields.z
+        # zeros all θ_l lower than boundry
+        @. p.soil.sub_sfc_scratch =
+            ClimaLand.heaviside.(
+                top_depth + sqrt(eps(FT)),
+                p.soil.sub_sfc_scratch,
+            ) * effective_saturation(ν, p.soil.θ_l, θ_r)
+        ClimaCore.Operators.column_integral_definite!(
+            ∫H_S_e_dz,
+            p.soil.sub_sfc_scratch,
+        )
+    else
+        # in the case where no layer is centered above boundry, use the top layer
+        ∫H_dz = soil_domain.fields.Δz_top
+        @. p.soil.sfc_scratch = soil_domain.fields.Δz_top + sqrt(eps(FT))
+        @. p.soil.sub_sfc_scratch =
+            soil_domain.fields.z_sfc - soil_domain.fields.z
+        # zeros all θ_l other than surface
+        @. p.soil.sub_sfc_scratch =
+            ClimaLand.heaviside.(p.soil.sfc_scratch, p.soil.sub_sfc_scratch) *
+            effective_saturation(ν, p.soil.θ_l, θ_r)
+        ClimaCore.Operators.column_integral_definite!(
+            ∫H_S_e_dz,
+            p.soil.sub_sfc_scratch,
+        )
+        # here ∫H_S_e_dz is equivalent to the effective_saturation of the top layer
```

Comment:
I tried using the `obtain_surface_space` and `top_face_to_surface` functions to do this. Both errored with certain domains because `level` did not work with them. Do you have any suggestions on how to get the values of the first layer?

Reply:
We dont want to use the value of the first layer because it is resolution dependent. Taking the average value over a predefined thickness is better defined (even though for coarse resolution sims, would amount to just the top layer, maybe).

Reply:
Of course, the albedo is probably only determined by the top ~cm or two, which does coincide with the top layer. But what if someone chooses a much coarser or finer resolution? It's good to put in some protection for that to make our simulation less resolution dependence.

If we do not want to support this now, top_center_to_surface should work, e.g.: https://github.com/CliMA/ClimaLand.jl/blob/e2050188bfffa9fe2ac8b581165d5fb2927ffb1a/src/standalone/Soil/boundary_conditions.jl#L235

Reply:
When the resolution is fine the function takes the average of all layers centered above the boundary. When the resolution is so coarse that no layer is centered above the boundary, the first layer is used. Isn't that what we want?

Reply:
That is what we want. I thought Gabriele was suggesting using the top layer value.